### PR TITLE
Notebooks updated

### PR DESCRIPTION
**Pull request by a-pouplin**
## What is the goal of this PR?

The notebooks have been updated and they are now working after the refactoring. 

## What are the changes implemented in this PR?

The modifications are the following: 
  - The imported functions have been renamed and the introductory notebook has been simplified,
  - The notebooks should produce plots that illustrate the AL package,
  - Those changes will not affect other files.

### CODE REVIEW

File: examples/notebooks/introduction.ipynb [hunk 1]
```diff
@@ -8,10 +8,16 @@
...
    "source": [
-    "[Active learning]( https://en.wikipedia.org/wiki/Active_learning_(machine_learning) ) is method that aims to query new data to be labelled, to improve the machine learning model. We suppose that we have a large pool of unlabelled data. Based on a predefined heuristic strategy, an Oracle will select a certain number of data `num_annotate` that will be annotated and used to re-train the model. Step after step, we should see an improvement in the model's performances. A baseline for the strategy would be to take the data randomly. "
+    "[Active learning]( https://en.wikipedia.org/wiki/Active_learning_(machine_learning) ) is a method that aims to collect and labelled new data in order to improve the machine learning model. Based on a predefined heuristic strategy, a certain number of data will be annotated and used to re-train the model. \n",
+    "In pyrelational, we use: \n",
+    "- an active learning ``strategy``, that query new data points based on specific selection criterion;\n",
+    "- a ``model``, written with pytorch lightning, that takes in input a neural networks and a set of arguments (ie: the number of epochs) used for training;\n",
```
Comment (thomasgaudelet) [inline examples/notebooks/introduction.ipynb:18]: this probably should be rephrased:

```suggestion
    "- a ``model manager`` that takes in input an uninstantiated ML model and a set of arguments (ie: the number of epochs) used for training;\n",
```

Comment (thomasgaudelet) [reply examples/notebooks/introduction.ipynb:18]: maybe giving more details about the arguments too, e.g. that there are used either to instantiate models or to perform training iterations.

Also might be worth mentioning the Oracle in the list here.

**Review by thomasgaudelet — CHANGES_REQUESTED** [review]

**Comment by paulmorio** [comment]
@a-pouplin thanks for the work on this! The package has been changed a little bit since this, most importantly the oracle being something that is instantiated and passed into the instantiation of the pipeline. You can see some of the examples in `examples/demo/` for this.

**Comment by a-pouplin** [comment]
Thank you for checking the notebooks ! I will update them today :)

File: pyrelational/strategies/regression/abstract_regression_strategy.py [hunk 2]
```diff
@@ -18,6 +19,8 @@ def __init__(self):
 
     def __call__(self, num_annotate: int, data_manager: DataManager, model: ModelManager) -> List[int]:
         output = self.train_and_infer(data_manager=data_manager, model=model)
-        scores = self.scoring_fn(x=output.squeeze(-1))
+        if isinstance(output, Tensor):
```
Comment (thomasgaudelet) [inline pyrelational/strategies/regression/abstract_regression_strategy.py:22]: in what instances is it not a Tensor?

Comment (a-pouplin) [reply pyrelational/strategies/regression/abstract_regression_strategy.py:22]: When we create a GP model or even a Bayesian neural network, the output can be a torch.distribution that contains the mean and the std. Depending on the strategy, we can use both. But a torch.distribution cannot be squeezed.

**Review by paulmorio — APPROVED** [review]
Thanks for the new updates, checked and approved (pending Thomas' note on the Tensor).

File: pyrelational/strategies/regression/abstract_regression_strategy.py (hunk 2 above)
Comment (thomasgaudelet) [reply pyrelational/strategies/regression/abstract_regression_strategy.py:22]: ah, of course! thanks for spotting it! There's a probably a few places in the strategies where this is a problem. We really need to handle multi-prediction then we wouldn't have to worry about this...

**Review by thomasgaudelet — APPROVED** [review]

**Review by paulmorio — APPROVED** [review]
LGTM!

**Pull request merged**